### DISCUSSION TITLE
v0.12.58 | Structured Logging

### DISCUSSION THREAD

**Discussion by MarvinJWendt** [pterm/pterm#481] [Announcements]
#### New Printer: Structured Logging!
<img width="500" src="https://raw.githubusercontent.com/pterm/pterm/master/_examples/logger/default/animation.svg" />

* feat(logger): added pretty structured logger by @MarvinJWendt in https://github.com/pterm/pterm/pull/480
* feat(rgb): made it possible to use RGB colors as background by @MarvinJWendt in https://github.com/pterm/pterm/pull/474
### Fixes 🔧
* fix(rgb): fixed a bug in RGB fading, where the `maxValue` would print with the `minValue` color. by @floaust in https://github.com/pterm/pterm/pull/479
### Other Changes
* examples: combine header examples by @MarvinJWendt in https://github.com/pterm/pterm/pull/467
* ci: added svgo workflow to optimize SVGs by @MarvinJWendt in https://github.com/pterm/pterm/pull/468
* docs(examples): added more examples by @MarvinJWendt in https://github.com/pterm/pterm/pull/469
* docs(README): improved readme by @MarvinJWendt in https://github.com/pterm/pterm/pull/476

**Full Changelog**: https://github.com/pterm/pterm/compare/v0.12.57...v0.12.58

<hr /><em>This discussion was created from the release <a href='https://github.com/pterm/pterm/releases/tag/v0.12.58'>v0.12.58 | Structured Logging</a>.</em>

**Comment by sheldonhull** [comment]
very nice! thanks for this 👍🏻

**Comment by ccremer** [comment]
A little while ago I made a bridge between pterm's beautiful color lines and the [go-logr](https://github.com/go-logr/logr) interface.
I named it Plogr: https://github.com/ccremer/plogr
(go-logr seems to be the slf4j in Java, just a facade where devs can choose their logging implementation)

However, this release's logs are much more readable, especially the key/values. So I thought to update Plogr with your implementation, but maybe it's a better idea to integrate the bridge for logr directly in this repo? WDYT @MarvinJWendt ?
